float-conversion compiler-error esp32 mpy1.13-103

Hello and thank you again for your work on this project. I can't stress how important this is to my project.

I've run into a compiler error when compiling the firmware on master (@c71920e):

```bash
CC /home/cal/uni/egh400/bdsim.micropython/firmware-build/ulab/code/numerical/numerical.c
/home/cal/uni/egh400/bdsim.micropython/firmware-build/ulab/code/numerical/numerical.c: In function 'numerical_median':
/home/cal/uni/egh400/bdsim.micropython/firmware-build/ulab/code/numerical/numerical.c:898:20: error: implicit conversion from 'mp_float_t {aka float}' to 'double' to match other operand of binary expression [-Werror=double-promotion]
             median *= 0.5;
                    ^
/home/cal/uni/egh400/bdsim.micropython/firmware-build/ulab/code/numerical/numerical.c:898:23: error: conversion to 'mp_float_t {aka float}' from 'double' may alter its value [-Werror=float-conversion]
             median *= 0.5;
                       ^
/home/cal/uni/egh400/bdsim.micropython/firmware-build/ulab/code/numerical/numerical.c:934:32: error: implicit conversion from 'mp_float_t {aka float}' to 'double' to match other operand of binary expression [-Werror=double-promotion]
                         median *= 0.5;
                                ^
/home/cal/uni/egh400/bdsim.micropython/firmware-build/ulab/code/numerical/numerical.c:934:35: error: conversion to 'mp_float_t {aka float}' from 'double' may alter its value [-Werror=float-conversion]
                         median *= 0.5;
                                   ^
cc1: all warnings being treated as errors
make: *** [../../py/mkrules.mk:63: build-GENERIC/code/numerical/numerical.o] Error 1
```

I've managed to fix the error by changing all "0.5"s to "0.5f"s, but I'm unsure if this is an adequate or necessary patch.

EDIT: I'm compiling micropython [v1.13-103 found here](https://github.com/micropython/micropython/commit/b137d064e9e0bfebd2a59a9b312935031252e742) as it resolves some issues with [micropython-stubber](https://github.com/BradenM/micropy-cli/issues/167).

--- a/code/numerical/numerical.c
+++ b/code/numerical/numerical.c
@@ -895,7 +895,7 @@ mp_obj_t numerical_median(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_
         if(!(len & 0x01)) { // len is an even number
             array -= ndarray->itemsize;
             median += ndarray_get_float_value(array, ndarray->dtype);
-            median *= 0.5;
+            median *= 0.5f;
         }
         return mp_obj_new_float(median);
     } else {
@@ -931,7 +931,7 @@ mp_obj_t numerical_median(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_
                     if(!(len & 0x01)) { // len is an even number
                         array -= ndarray->strides[ax];
                         median += ndarray_get_float_value(array, ndarray->dtype);
-                        median *= 0.5;
+                        median *= 0.5f;
                         array += ndarray->strides[ax];
                     }
                     array -= ndarray->strides[ax] * (len >> 1);

--- a/code/numerical/numerical.c
+++ b/code/numerical/numerical.c
@@ -895,7 +895,7 @@ mp_obj_t numerical_median(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_
         if(!(len & 0x01)) { // len is an even number
             array -= ndarray->itemsize;
             median += ndarray_get_float_value(array, ndarray->dtype);
-            median *= 0.5f;
+            median *= MICROPY_FLOAT_CONST(0.5);
         }
         return mp_obj_new_float(median);
     } else {
@@ -931,7 +931,7 @@ mp_obj_t numerical_median(size_t n_args, const mp_obj_t *pos_args, mp_map_t *kw_
                     if(!(len & 0x01)) { // len is an even number
                         array -= ndarray->strides[ax];
                         median += ndarray_get_float_value(array, ndarray->dtype);
-                        median *= 0.5f;
+                        median *= MICROPY_FLOAT_CONST(0.5);
                         array += ndarray->strides[ax];
                     }
                     array -= ndarray->strides[ax] * (len >> 1);

